tests/servo_test.py

```diff
@@ -539,10 +531,6 @@ def invalid_measure(self, invalid: str, another: int) -> None:
             pass
 
     assert error
-    assert (
-        str(error.value)
-        == """invalid before event handler "before:measure": encountered unexpected parameters "another and invalid" in callable signature "(self, invalid: str, another: int) -> None", expected "(self, *, metrics: 'List[str]' = None, control: 'servo.types.Control' = Control(duration=Duration('0'), delay=Duration('0'), warmup=Duration('0'), settlement=None, load=None, userdata=None, environment=None)) -> 'None'\""""
-    )
```

Comment:
> adjustment to tests (matching full text of error output is not a good behavior imho, matching the type is good enough)

I'm in full agreement here but all three of these tests raise the same type of error. Can you add some sort of check that the raised error is the one we were expecting? (for all three tests)

Reply:
The `with pytest.raises(TypeError) as error` above ensures we're getting the right type of error.

Reply:
Validating the type of error is not sufficient since the same type `TypeError` is raised by those tests/logic for different reasons. Please add a check that the error was raised for the expected reason so regressions that introduce unexpected `TypeError`s can be caught sooner than later